BUILDING.md

```diff
@@ -491,6 +492,22 @@ $ gdb /opt/node-debug/node core.node.8.1535359906
 $ backtrace
 ```
 
+#### Building an ASAN build
+
+[ASAN](https://github.com/google/sanitizers) can help detect various memory
+related bugs. ASAN builds are currently only supported on linux.
+If you want to check it on windows or macOS or you want a consistent toolchain
+on linux, you can try [docker](https://www.docker.com/products/docker-desktop)
+ (using Image like `gengjiawen/node-build:2020-02-14`)
+
+The `--debug` is not necessary, but it can show clear stacktrace
+if hit asan issue.
+
+``` console
+$  ./configure --debug --enable-asan && make -j8
```

Comment:
nit: seems like we use `-j4` as an example through the entire doc. Maybe we want to use it here too for consistency?